src/Block/MediaBlockService.php

```diff
@@ -209,10 +204,10 @@ protected function getFormatChoices(?MediaInterface $media = null): array
     protected function getMediaBuilder(FormMapper $form): FormBuilderInterface
     {
         // simulate an association ...
-        $fieldDescription = $this->getMediaAdmin()->getModelManager()->getNewFieldDescriptionInstance($this->mediaAdmin->getClass(), 'media', [
+        $fieldDescription = $this->fieldDescriptionFactory->create($this->mediaAdmin->getClass(), 'media', [
             'translation_domain' => 'SonataMediaBundle',
         ]);
-        $fieldDescription->setAssociationAdmin($this->getMediaAdmin());
+        $fieldDescription->setAssociationAdmin($this->mediaAdmin);
         $fieldDescription->setAdmin($form->getAdmin());
```

Comment:
The AdminBundle FormMapper has this method, but not the blockBundle one...

What I don't understand is that I don't find class which implements the FormMapper interface of BlockBundle.

Reply:
I think this wouldn't be a problem using the admin createFieldDescription